pandas/core/frame.py

```diff
@@ -685,6 +685,8 @@ def __init__(
                 # INFO(ArrayManager) by default copy the 2D input array to get
                 # contiguous 1D arrays
                 copy = True
+            elif using_copy_on_write() and isinstance(data, np.ndarray):
```

Comment:
Potential alternative is to express this in the inverse: `.. and not isinstance(data, (Series, DataFrame))`. 

That would ensure that also other "array-likes" that would coerce zero-copy into a numpy array but are not exactly np.ndarray would get copied by default (not fully sure how our constructor currently handles such array like objects, though). 
I think in the end only for pandas objects we can keep track of references, so only in that case the default should be a shallow copy?